tests/components/notion/test_config_flow.py

```diff
@@ -42,45 +27,88 @@ async def test_duplicate_error(hass):
 
-    result = await flow.async_step_user(user_input=None)
 
+async def test_step_reauth(hass):
+    """Test that the reauth step works."""
+    MockConfigEntry(
+        domain=DOMAIN,
+        unique_id="user@email.com",
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_REAUTH},
+        data={CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"},
+    )
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
     assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
+    assert result["step_id"] == "reauth_confirm"
+
+    with patch("homeassistant.components.notion.config_flow.async_get_client"), patch(
+        "homeassistant.components.notion.async_setup_entry", return_value=True
+    ), patch("homeassistant.config_entries.ConfigEntries.async_reload"):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={CONF_PASSWORD: "password"}
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+        assert result["reason"] == "reauth_successful"
+
+    assert len(hass.config_entries.async_entries()) == 1
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    with patch("homeassistant.components.notion.config_flow.async_get_client"):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == "user"
 
 
-async def test_step_user(hass, mock_aionotion):
+async def test_step_user(hass):
     """Test that the user step works."""
     conf = {CONF_USERNAME: "user@host.com", CONF_PASSWORD: "password123"}
 
-    flow = config_flow.NotionFlowHandler()
-    flow.hass = hass
-    flow.context = {"source": SOURCE_USER}
-
-    result = await flow.async_step_user(user_input=conf)
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == "user@host.com"
-    assert result["data"] == {
-        CONF_USERNAME: "user@host.com",
-        CONF_PASSWORD: "password123",
-    }
+    with patch(
+        "homeassistant.components.notion.async_setup_entry", return_value=True
+    ), patch("homeassistant.components.notion.config_flow.async_get_client"):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}, data=conf
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "user@host.com"
+        assert result["data"] == {
+            CONF_USERNAME: "user@host.com",
+            CONF_PASSWORD: "password123",
+        }
```

Comment:
Config entry is set up in parallel, so you need to wait till that task is done before patching out async_setup_entry to avoid potential flaky test.

```suggestion
        }
        await hass.async_block_till_done()
```